src/main/common/filter.c

```diff
@@ -38,65 +44,61 @@ float filterApplyPt1(float input, filterStatePt1_t *filter, uint8_t f_cut, float
- */
-static int8_t gyroFIRCoeff_500[3][FILTER_TAPS] = { { 0, 18, 14, 16, 20, 22, 24, 25, 25, 24, 20, 18, 12, 18 },  // TODO - NEEDS TABLE
-                                                  { 23, 12, 16, 18, 20, 20, 23, 20, 22, 18, 16, 14, 12, 22 },  //
-                                                  { 36, 12, 14, 14, 16, 16, 18, 18, 18, 16, 16, 14, 12, 36 } };//
-static int8_t gyroFIRCoeff_1000[3][FILTER_TAPS] = { { 0, 0, 0, 0, 0, 0, 0, 12, 23, 40, 51, 52, 40, 38 }, // looptime=1000; group delay 2.5ms; -0.5db = 32Hz ; -1db = 45Hz; -5db = 97Hz; -10db = 132Hz
-                                                  { 0, 0, 0, 0, 0, 18, 30, 42, 46, 40, 34, 22, 8, 8},    // looptime=1000; group delay 3ms;   -0.5db = 18Hz ; -1db = 33Hz; -5db = 81Hz; -10db = 113Hz
-                                                  { 0, 0, 0, 0, 0, 18, 12, 28, 40, 44, 40, 32, 22, 20} };// looptime=1000; group delay 4ms;   -0.5db = 23Hz ; -1db = 35Hz; -5db = 75Hz; -10db = 103Hz
-static int8_t gyroFIRCoeff_2000[3][FILTER_TAPS] = { { 0, 0, 0, 0, 0, 0, 0, 0, 6, 24, 58, 82, 64, 20 },   // looptime=2000, group delay 4ms;   -0.5db = 21Hz ; -1db = 31Hz; -5db = 71Hz; -10db = 99Hz
-                                                  { 0, 0, 0, 0, 0, 0, 0, 14, 22, 46, 60, 56, 40, 24},    // looptime=2000, group delay 5ms;   -0.5db = 20Hz ; -1db = 26Hz; -5db = 52Hz; -10db = 71Hz
-                                                  { 0, 0, 0, 0, 0, 14, 12, 26, 38, 44, 42, 34, 24, 20} };// looptime=2000, group delay 7ms;   -0.5db = 11Hz ; -1db = 18Hz; -5db = 38Hz; -10db = 52Hz
-static int8_t gyroFIRCoeff_3000[3][FILTER_TAPS] = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 36, 88, 88, 44 },    // looptime=3000, group delay 4.5ms; -0.5db = 18Hz ; -1db = 26Hz; -5db = 57Hz; -10db = 78Hz
-                                                  { 0, 0, 0, 0, 0, 0, 0, 0, 14, 32, 64, 72, 54, 28},     // looptime=3000, group delay 6.5ms; -0.5db = 16Hz ; -1db = 21Hz; -5db = 42Hz; -10db = 57Hz
-                                                  { 0, 0, 0, 0, 0, 0, 6, 10, 28, 44, 54, 54, 38, 22} };  // looptime=3000, group delay 9ms;   -0.5db = 10Hz ; -1db = 13Hz; -5db = 32Hz; -10db = 45Hz
-
-int8_t * filterGetFIRCoefficientsTable(uint8_t filter_level, uint16_t targetLooptime)
+/* sets up a biquad Filter */
+void BiQuadNewLpf(uint8_t filterCutFreq, biquad_t *newState, float refreshRate)
 {
-    if (filter_level == 0) {
-        return NULL;
+	float samplingRate;
+
+    if (!refreshRate) {
+        samplingRate = 1 / (targetLooptime * 0.000001f);
+    } else {
+        samplingRate = refreshRate;
     }
 
-    int firIndex = constrain(filter_level, 1, 3) - 1;
+    float omega, sn, cs, alpha;
+    float a0, a1, a2, b0, b1, b2;
 
-    // For looptimes faster than  700 use filter for 2kHz looptime
-    if (targetLooptime < 700) {
-        return gyroFIRCoeff_500[firIndex];
-    }
+    /* setup variables */
+    omega = 2 * M_PI_FLOAT * (float) filterCutFreq / samplingRate;
+    sn = sinf(omega);
```

Comment:
Why use libm accurate implementation here? This will negate the effort of
saving some flash space by approximate implementation.